File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        graph={} 
        indegree={} 
        for i in range(n): 
            indegree[i]=0 
            graph[i]=0
        for x,y in edges:
            graph[y]+=1 
        vert=[] 
        for i in range(n):
            if graph[i]==0:
                vert.append(i)
        return vert

File: 1557-minimum-number-of-vertices-to-reach-all-nodes/1557-minimum-number-of-vertices-to-reach-all-nodes.cpp
class Solution {
public:
    vector<int> findSmallestSetOfVertices(int n, vector<vector<int>>& edges) { 
        map<int,int> m;
        for(int i=0;i<edges.size();i++) 
        {
            m[edges[i][1]]++;
        }
        
        vector<int> res;
        for(int i=0;i<n;i++)
        {  
            if(m[i]==0) 
                res.push_back(i);
        }
        return res;
        
    }
};